core/testcontainers/core/config.py

```diff
@@ -131,6 +131,10 @@ def tc_properties_get_tc_host(self) -> Union[str, None]:
     def timeout(self) -> int:
         return self.max_tries * self.sleep_time
 
+    @property
+    def ryuk_docker_socket(self) -> str:
+        return get_docker_socket()
```

Comment:
```suggestion
    @property
    def ryuk_docker_socket(self) -> str:
        if not self._ryuk_docker_socket:
            self.ryuk_docker_socket  = get_docker_socket()
        return self._ryuk_docker_socket

    @ryuk_docker_socket.setter
    def ryuk_docker_socket(self, value: str) -> None:
        self._ryuk_docker_socket = value
```
Cache the result but make it changeable